firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -2036,8 +2036,10 @@ class BrowserViewController: UIViewController,
         let alwaysShowSearchSuggestionsView = browserViewControllerState?
             .searchScreenState
             .showSearchSugestionsView ?? false
-        let isSettingEnabled = profile.searchEngines.isPrivateModeSettingEnabled
-
+        // swiftlint:disable line_length
+        let isSettingEnabled = featureFlags.isFeatureEnabled(.firefoxSuggestFeature, checking: .buildAndUser) ?         profile.searchEngines.isPrivateModeSettingEnabled :
+            profile.searchEngines.shouldShowPrivateModeSearchSuggestions
+        // swiftlint:enable line_length
```

Comment:
I would argue this is not a great use of disabling a swiftlint rule. It's there to ensure readabiliity, and we're cramming things just to use a ternary which feels counter-intuitive.

Also, weird spacing! lolololol

You can do something like

```suggestion
        let nimbusSuggestEnabled = featureFlags.isFeatureEnabled(.firefoxSuggestFeature, checking: .buildAndUser)
        let engines = profile.searchEngines
        let isSettingEnabled = nimbusSuggestEnabled ? engines.isPrivateModeSettingEnabled :
            engines.shouldShowPrivateModeSearchSuggestions
```

Or you can do one of the new fancy `switch` uses

```suggestion
        let isSettingEnabled = switch featureFlags.isFeatureEnabled(.firefoxSuggestFeature, checking: .buildAndUser) {
        case true: profile.searchEngines.isPrivateModeSettingEnabled
        case false: profile.searchEngines.shouldShowPrivateModeSearchSuggestions
```

or the `if` way or whatever.